README.md

```diff
@@ -6,10 +6,68 @@
+The Frequenz Microgrid API is an API for managing and monitoring components
+within a microgrid. It offers various functionalities related to microgrid
+management, including metadata retrieval, component listing, setting power
+levels, and controlling the state of components. The API is structured around
+the concept of a microgrid, which is a localized energy system composed of
+various interconnected components.
 
-This repository provides the core gRPC and protobuf spec together with
-support for generating dedicated API packages for various languages.
+
+## Objective
+
+The objective of this API is to provide a standardized interface for
+interacting with and controlling components within a microgrid. It aims to
+enable efficient management and monitoring of the microgrid's components,
+including batteries, inverters, chargers, meters, and more. It facilitates
+tasks such as retrieving metadata about the microgrid, filtering and listing
+components, managing component power levels, and controlling component
+states.
+
+### Key Features
+
+- **Metadata Retrieval:** The API allows users to retrieve metadata about the
+microgrid, providing information about the microgrid's unique identifier (ID)
+and geographical coordinates.
+
+- **Component Listing:** Users can list components in the microgrid, optionally
+filtered by component IDs and categories.
+
+- **Connection Listing:** The API provides a way to list connections between
+components. Connections are represented as pairs of component IDs, describing
+the flow of current according to the passive sign convention.
+
+- **Power Level Control:** Users can set the active and reactive power levels of
+components that support it.
+- **Component State Control:** For components that support it, the API supports
+starting, stopping, and transitioning components between these and the
+standby states.
+
+- **Bounds Management:** Users can set exclusion and inclusion bounds for specific
+metrics of a component. These bounds define the acceptable and unacceptable
+ranges of metric values.
+
+### Example Use Cases:
+
+This API is designed for use in microgrid management and control systems. It
+can be applied in various scenarios, including:
+
+- **Microgrid Monitoring:** Monitoring the status and performance of components in
+a microgrid, allowing operators to ensure efficient energy distribution.
+
+- **Optimizing Energy Storage:** Managing the charge and discharge power of
+batteries to optimize energy storage and consumption within the microgrid.
+
+- **Performance Enhancements:** Setting exclusion bounds for metrics to enhance
+system performance and prevent extreme values in battery SoCs or power
+output.
+
+### Target Audience:
+
+This API is designed for application developers in the energy sector who
+focus on the tasks of optimizing microgrid electricity flows. Its design aims
+to be as developer-friendly as possible, requiring no prior knowledge in
```

Comment:
prior knowledge of any specific hardware implementation would be better phrased.

Reply:
Done.